astronomer/providers/great_expectations/operators/great_expectations.py

```diff
@@ -0,0 +1,375 @@
+import os
+from datetime import datetime
+from typing import Any, Callable, Dict, Optional, Union
+
+import great_expectations as ge
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+from airflow.models import BaseOperator
+from great_expectations.checkpoint import Checkpoint
+from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+from great_expectations.data_context import BaseDataContext
+from great_expectations.data_context.types.base import (
+    BaseStoreBackendDefaults,
+    CheckpointConfig,
+    DataContextConfig,
+    FilesystemStoreBackendDefaults,
+)
+from great_expectations.data_context.util import instantiate_class_from_config
+
+
+class GreatExpectationsOperator(BaseOperator):
+    """
+    An operator to leverage Great Expectations as a task in your Airflow DAG.
+
+    Current list of expectations types:
+    https://docs.greatexpectations.io/en/latest/reference/glossary_of_expectations.html
+
+    How to create expectations files:
+    https://docs.greatexpectations.io/en/latest/guides/tutorials/how_to_create_expectations.html
+
+    :param run_name: Identifies the validation run (defaults to timestamp if not specified)
+    :type run_name: Optional[str]
+    :param conn_id: The name of a connection in Airflow
+    :type conn_id: Optional[str]
+    :param base_directory: The directory an InferredAssetFilesystemDataConnector will search for files under
+    :type base_directory: Optional[str]
+    :param file_regex: The regex used to load files with an InferredAssetFilesystemDataConnector
+    :type file_regex: Optional[Dict]
+    :param execution_engine: The execution engine to use when running Great Expectations
+    :type execution_engine: Optional[str]
+    :param store_backend_defaults: Determines how backend defaults are stored
+    :type store_backend_defaults: Optional[BaseStoreBackendDefaults]
+    :param batch_kwargs: Key-word arguments for a batch request
+    :type batch_kwargs: Optional[Dict]
```

Comment:
Gotcha - this would be good to discuss. The current Operator doesn't use this, and the V2 version of the Operator might have, but this wouldn't work together with that.